### DISCUSSION TITLE
Expand folders that have differences only

### DISCUSSION THREAD

**Discussion by Xevel** [WinMerge/winmerge#1381] [Ideas]
Hello,

In Tree view, there is a command that will expand all sub directories, but I did not find a command that would expand only the ones with differences.
It seems to me that is would be very useful: the tool is there to help find differences, and after scanning the whole tree it knows where they are.
But right now we have either to click many times to get to where they are "smartly" (without opening folders without changes to avoid crowding the screen) - or expand everything and be rewarded with potentially most of the information being useless (hundreds upon hundreds of folders opened to show files that are identical...), making navigation harder (since there is no way to jump to next different file, only "next difference", which will stop at each folders in the path of a different file).

Cheers,
Nicolas

**Comment by sdottaka** [comment]
Unfortunately, it is currently not possible to expand only the folders with differences.

However, if you uncheck Show Identical Items menu item, the identical folders will be hidden, so how about clicking Expand All Subfolders menu item after that?

![image](https://user-images.githubusercontent.com/98126/175936378-fd9a8c49-8da4-48a7-a57e-2c3e05d342b6.png)

**Reply by Xevel** [reply]
That seems like an acceptable workaround for most cases, thank you.
Still not perfect as I would want to see also the identical files in the folders where there are files with differences (to be able to open them from that interface as the differences might have something to do with these other files that have not been changed), but a step in the right direction for me.

Cheers,
Nicolas

**Comment by sdottaka** [comment]
Implemented In PR #1964
